api/v1alpha3/machinehealthcheck_webhook.go

```diff
@@ -90,14 +90,22 @@ func (m *MachineHealthCheck) validate(old *MachineHealthCheck) error {
 	var allErrs field.ErrorList
 
 	// Validate selector parses as Selector
-	_, err := metav1.LabelSelectorAsSelector(&m.Spec.Selector)
+	selector, err := metav1.LabelSelectorAsSelector(&m.Spec.Selector)
 	if err != nil {
 		allErrs = append(
 			allErrs,
 			field.Invalid(field.NewPath("spec", "selector"), m.Spec.Selector, err.Error()),
 		)
 	}
 
+	// Validate that the selector isn't empty.
```

Comment:
nit: Since empty label selector is actually a valid semantic https://github.com/kubernetes/kubernetes/blob/665e76d97623447d13bb3b33b68591a985b49c9d/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go#L1041
It'd be helpful to add the reasoning behind this in the commit description (`git co -m"" -m"This is to prevent users from accidentally..."`).